scripts/compress_images.py

```diff
@@ -0,0 +1,255 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to compress images in the repository using GraphicsMagick."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+
+from PIL import Image
+from typing import List, TypedDict, Union
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+class ImageCompressor:
+    """A class to handle image compression using GraphicsMagick."""
+
+    IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION = {'.png'}
+    IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION = {'.jpg', '.jpeg', '.webp'}
+    ALL_IMAGES_EXTENSIONS = (
+        IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION |
+        IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION
+    )
+    TOLERANCE = 0.99
+    OUTPUT_DIR = 'compressed_images'
+
+    def __init__(
+        self,
+        input_path: Union[str, pathlib.Path],
+        output_dir: str = OUTPUT_DIR,
+        tolerance: float = TOLERANCE
+    ):
+        """Initialize the ImageCompressor.
+
+        Args:
+            input_path: str. Images directory.
+            output_dir: str. Directory to store compressed images.
+            tolerance: float. Compression threshold.
+        """
+        self.input_path = pathlib.Path(input_path)
+        self.output_dir = output_dir
+        self.tolerance = tolerance
+        self.logger = logging.getLogger(__name__)
+
+    def get_compression_type(self, file_extension: str) -> str:
+        """Determine compression type based on file extension.
+
+        Args:
+            file_extension: str. The file extension of the image.
+
+        Returns:
+            type: str. Compression type (Zip or LZW).
+        """
+        return (
+            'Zip' if file_extension in 
+            self.IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION
+            else 'LZW'
+        )
+
+    def run_compression_logic(
+            self,
+            file_path: pathlib.Path,
+            output_file_path: pathlib.Path,
+    ) -> List[CompressedImageInfo] | None:
+        """Helper function for running compression logic.
+
+        Args:
+            file_path: pathlib.Path. The path to the image file.
+            output_file_path: pathlib.Path. The path to the output file.
+
+        Returns:
+            result_image: List[CompressedImageInfo]. List of all compressible
+            images over repository with attributes as path, original_size
+            and new_size.
+        """
+
+        file_extension = file_path.suffix.lower()
+        result_image: List[CompressedImageInfo] = []
```

Comment:
variable name is confusing since this isn't an image. Maybe `image_compressibility_dicts`?